### PR TITLE
chore: upgrade rust version to 1.82

### PR DESCRIPTION
**Pull request by DSharifi**
This PR upgrades the rust version of the repo to 1.82.
All changes made are to satisfy new clippy lints.

The PR also disables the `TLA` tool per instruction from @oggy-dfin, as it was running out of memory. See [BB link.](https://dash.zh1-idx1.dfinity.network/invocation/118e584d-c226-4dd6-830c-95d06aad81de?target=%2F%2Frs%2Fnns%2Fgovernance%3Agovernance_integration_test_tests%2Fgovernance&targetStatus=6#@7066)

### CODE REVIEW

File: rs/ledger_suite/icrc1/ledger/canbench_results/canbench_u256.yml [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
-        instructions: 126706
+        instructions: 124353
         heap_increase: 129
         stable_memory_increase: 129
       upgrade:
-        instructions: 8724099
+        instructions: 8722103
         heap_increase: 258
         stable_memory_increase: 129
 version: 0.1.7
```
Comment (DSharifi) [inline rs/ledger_suite/icrc1/ledger/canbench_results/canbench_u256.yml:42]: I updated the expected results with `bazel run //rs/ledger_suite/icrc1/ledger:canbench_u64_update` due to the canbench failing on CI. 
See: https://dash.zh1-idx1.dfinity.network/invocation/5635b4dd-0d51-4b38-bef7-cf705bf43f00

**Comment by DSharifi** [comment]
I updated the expected canbench results for icrc1 as it was failing on CI:
See: https://dash.zh1-idx1.dfinity.network/invocation/5635b4dd-0d51-4b38-bef7-cf705bf43f00

**Review by marko-k0 — APPROVED** [review]

**Review by oggy-dfin — APPROVED** [review]

File: rs/nns/governance/feature_flags.bzl [hunk 2]
```diff
@@ -5,6 +5,6 @@ Feature flags for turning TLA-based checks on and off.
 def test_with_tla():
     return select({
         # Remove "tla" here to disable TLA-related checks in the CI
```
Comment (aterga) [inline rs/nns/governance/feature_flags.bzl:7]: Could you explain why you're disabling TLA here? (I know it's failing the pipeline, but is there nothing better that can be done in this case?)

```suggestion
        # Add "tla" here to enable TLA-related checks in the CI
```

**Review by aterga — APPROVED** [review]

**Review by mbjorkqvist — APPROVED** [review]
Thanks @DSharifi !

**Review by maksymar — APPROVED** [review]
LGTM for execution part

**Pull request merged**